pxr/usd/usdLightField/schema.usda

```diff
@@ -0,0 +1,338 @@
+
+             Attributes are provided in both `float` and `half` types for some
+             easy data footprint affordance, data consumers should prefer
+             `float` version if available.
+             """
+    inherits = </APISchemaBase>
+    customData = {
+        token apiSchemaType = "singleApply"
+        token[] apiSchemaCanOnlyApplyTo = ["ParticleField"]
+    }
+)
+{
+    float3[] scales (
+        doc = """Affine scale factor applied to the base shape defined in the
+                'shape' attribute."""
+    )
+    half3[] scalesh (
+        doc = """Affine scale factor applied to the base shape defined in the
+                'shape' attribute. If the float precision version is available
+                it should be preferred."""
+    )
+}
+
+class "OpacityAttributeAPI" (
+    doc = """A ParticleField related appliedAPI schema that provides a
+             opacity attribute to define the opacity of the particles.
+
+             The opacity here should be in the range [0, 1], and inline with the
+             traditional sense of computer graphics opacity, not the
+             transformed data sometimes seen in PLY files associated with
+             gaussian splats, where the values need to be processed with a
+             sigmoid activation function.
+
+             Attributes are provided in both `float` and `half` types for some
+             easy data footprint affordance, data consumers should prefer
+             `float` version if available.
+             """
+    inherits = </APISchemaBase>
+    customData = {
+        token apiSchemaType = "singleApply"
+        token[] apiSchemaCanOnlyApplyTo = ["ParticleField"]
+    }
+)
+{
+    float[] opacities (
+        doc = "Opacity for each particle."
+    )
+    half[] opacitiesh (
+        doc = """Opacity for each particle. If the float precision version is
+                available it should be preferred."""
+    )
+}
+
+class "GaussianShapeAPI" (
+    doc = """Defines the Gaussian shape function.
+    This is really just a spherical region the shape of which is reshaped by
+    the associated scale data source."""
+    inherits = </APISchemaBase>
+    customData = {
+        token apiSchemaType = "singleApply"
+        token[] apiSchemaCanOnlyApplyTo = ["ParticleField"]
+    }
+)
+{
+}
+
+# This is an example of an alternate shape. No concrete use-case for this
+# shape yet.
+class "TriangleShapeAPI" (
+    doc = """Defines a base shape as a equilateral triangle with given edge
+            length."""
+    inherits = </APISchemaBase>
+    customData = {
+        token apiSchemaType = "singleApply"
+        token[] apiSchemaCanOnlyApplyTo = ["ParticleField"]
+    }
+)
+{
+    float kernelShape:triangle:edgeLength = 1 (
+        doc = """The edge length of the equilateral triangle before any scaling
+                is applied."""
+    )
+}
+
+
+class "GaussianFalloffFunctionAPI" (
+    doc = """Defines a Gaussian fall-off function. An opacity data source is
```

Comment:
tricky interplay between schemas, here, and illustrates the need for and kinds of validation we'll want.